pkg/pub_package_reader/lib/pub_package_reader.dart

```diff
@@ -307,6 +308,27 @@ Iterable<ArchiveIssue> checkAuthors(String pubspecContent) sync* {
   }
 }
 
+final _preDart3 = VersionConstraint.parse('<3.0.0');
+final _firstDart3Pre = Version.parse('3.0.0').firstPreRelease;
+
+/// Checks if the version range is acceptable by current SDKs.
+Iterable<ArchiveIssue> checkSdkVersionRange(Pubspec pubspec) sync* {
+  final sdk = pubspec.environment?['sdk'];
+  if (sdk == null) return;
+  // We still allow pubspec without SDK constraints.
+  if (sdk.isAny) return;
```

Comment:
This is only relevant for Dart 1.x, and there is afaik no reason they couldn't be published with an SDK constraint.

Should we not just force that upon Dart 1.x users... We could do a quick scan of pubspec published past year to see if anyone still does this..

feel free to just file that as a separate issue, I see no reason to block this...

Reply:
Updated validation to require min and max ranges. PTAL.